[BUG] test_window_aggregate_udf_array_from_python fails on Spark 3.2.0

**Describe the bug**
```bash
=========================== short test summary info ============================

FAILED ../../src/main/python/udf_test.py::test_window_aggregate_udf_array_from_python[Lower_Upper-Short][DATAGEN_SEED=1743009810, TZ=Canada/Newfoundland, INJECT_OOM, IGNORE_ORDER] - py4j.protocol.Py4JJavaError: An error occurred while calling o1829246.collectToPython.
org.apache.spark.SparkException: Job 18254 cancelled RAPIDS Integration Test Job 18254 exceeded the timeout of 30 seconds, cancelling. Look into fixing the test or reducing its execution time. If necessary, adjust the timeout using the marker pytest.mark.spark_job_timeout(seconds,dump_threads)
```

It seems this test is failing with the timeout added in #12346.

--- a/integration_tests/src/main/python/udf_test.py
+++ b/integration_tests/src/main/python/udf_test.py
@@ -21,12 +21,6 @@ from spark_session import (is_before_spark_330, is_before_spark_331, is_before_s
 
 from pyspark.sql.pandas.utils import require_minimum_pyarrow_version, require_minimum_pandas_version
 
-# TODO undo once we fix 400 and 350db143
-# https://github.com/NVIDIA/spark-rapids/pull/12383
-pytestmark = [
-    pytest.mark.spark_job_timeout(seconds=30)
-]
-
 try:
     require_minimum_pandas_version()
 except Exception as e:

